shpc/main/modules/templates/docker.tcl

```diff
@@ -74,24 +76,32 @@ set-alias {|module_name|}-shell "${shellCmd}"
 {% if aliases %}
 if { [ module-info shell bash ] } {
   if { [ module-info mode load ] } {
-{% for alias in aliases %}    puts stdout "function {{ alias.name }}() { ${execCmd} {% if alias.docker_options %} {{ alias.docker_options | replace("$", "\$") }} {% endif %} --entrypoint {{ alias.entrypoint | replace("$", "\$") }} ${containerPath} {{ alias.args | replace("$", "\$") }} \$@; }; export -f {{ alias.name }};"
+{% for alias in aliases %}    puts stdout "function {{ alias.name }}() { ${execCmd} {% if alias.docker_options %} {{ alias.docker_options | replace("$", "\$") }} {% endif %} --entrypoint {{ alias.entrypoint | replace("$", "\$") }} ${containerPath} {{ alias.command | replace("$", "\$") }} \"\$@\"; }; export -f {{ alias.name }};"
 {% endfor %}
   }
   if { [ module-info mode remove ] } {
 {% for alias in aliases %}    puts stdout "unset -f {{ alias.name }};"
 {% endfor %}
   }
 } else {
-{% for alias in aliases %}  set-alias {{ alias.name }} "${execCmd} {% if alias.docker_options %} {{ alias.docker_options | replace("$", "\$") }} {% endif %} --entrypoint {{ alias.entrypoint | replace("$", "\$") }} ${containerPath} {{ alias.args | replace("$", "\$") }}"
+{% for alias in aliases %}  set-alias {{ alias.name }} "${execCmd} {% if alias.docker_options %} {{ alias.docker_options | replace("$", "\$") }} {% endif %} --entrypoint {{ alias.entrypoint | replace("$", "\$") }} ${containerPath} {{ alias.command | replace("$", "\$") }}"
 {% endfor %}
 }
 {% endif %}
 
 # A customizable exec function
-set-alias {|module_name|}-exec "${execCmd} --entrypoint \"\" ${containerPath}"
+if { [ module-info shell bash ] } {
```

Comment:
This makes sense! And I remember the empty entrypoint was needed for it to work.